ImageProcessing/src/BlueFramework/ImageProcessing/Color.h

```diff
@@ -333,43 +314,43 @@ struct Color<T, 3> {
 
 	// get the X-Value of XYZ from sRGB value
 	double getXYZ_X() {
-		int red = (*this)[0];
-		int green = (*this)[1];
-		int blue = (*this)[2];
+		const int red = (*this)[0];
+		const int green = (*this)[1];
+		const int blue = (*this)[2];
 
-		double r = inverseGammaEncoding((double)red / 255);
-		double g = inverseGammaEncoding((double)green / 255);
-		double b = inverseGammaEncoding((double)blue / 255);
+		const double r = inverseGammaEncoding(static_cast<double>(red) / 255);
+		const double g = inverseGammaEncoding(static_cast<double>(green) / 255);
+		const double b = inverseGammaEncoding(static_cast<double>(blue) / 255);
 		return 0.4124564 * r + 0.3575761 * g + 0.1804375 * b;
 	}
 
 	// get the Y-Value of XYZ from sRGB value
 	double getXYZ_Y() {

```

Comment:
these function can be declared `const` as well

Reply:
Has also been changed to const.